### PR TITLE
use squashfuse_ll if it is available.

### PR DESCRIPTION
**Pull request by smoser**
feat: Use squashfuse_ll if available for faster squashfs.
    
squashfuse README says of squashfuse_ll:
    
>   Like `squashfuse`, but implemented using the low-level
>   FUSE API. It's a tiny bit faster, but less portable.
    
But some reports say things like "30x times faster."
https://github.com/snapcore/snapd/pull/7477

### CODE REVIEW

File: squashfs/squashfs.go [hunk 1]
```diff
@@ -168,7 +168,15 @@ func ExtractSingleSquash(squashFile string, extractDir string, storageType strin
...
 
-	if p := which("squashfuse"); p != "" {
+	if realuid := os.Getenv("STACKER_REAL_UID"); realuid == "0" {
+		log.Debugf("Extracting %s -> %s with kernel mount", squashFile, extractDir)
+		cmd := exec.Command("mount", "-oloop,ro", squashFile, extractDir)
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		cmd.Stdin = nil
+		return cmd.Run()
+		return cmd.Start()
```
Comment (hallyn) [inline squashfs/squashfs.go:187]: Two returns in a row?

File: squashfs/squashfs.go [hunk 2]
```diff
@@ -176,21 +185,21 @@ func ExtractSingleSquash(squashFile string, extractDir string, storageType strin
...
-	} else if p := which("squashfuse"); p != "" {
+	} else if sqfuse := findSqfusePath(); sqfuse != "" {
 		// given extractDir of path/to/some/dir[/], log to path/to/some/.dir-squashfs.log
 		extractDir := strings.TrimSuffix(extractDir, "/")
 
 		var cmdOut io.Writer
 		logf := path.Join(path.Dir(extractDir), "."+path.Base(extractDir)+"-squashfuse.log")
 		if cmdOut, err = os.OpenFile(logf, os.O_RDWR|os.O_TRUNC|os.O_CREATE, 0644); err != nil {
-			log.Infof("Failed to open %s for write: %v", p, err)
+			log.Infof("Failed to open %s for write: %v", logf, err)
```
Comment (hallyn) [inline squashfs/squashfs.go:195]: the logf fix is needed regardless, independent of this commit, right?

Comment (smoser) [reply squashfs/squashfs.go:195]: Yes

**Review by hallyn — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**